torch/csrc/autograd/functions/batch_normalization.cpp

```diff
@@ -206,7 +206,7 @@ auto BatchNormBackward::apply(const variable_list& grad_outputs) -> variable_lis
                                  std::move(grad_bias));
   return wrap_outputs(all_inputs, std::move(outputs), [&](FunctionFlags f) {
     return std::make_shared<BatchNormBackwardBackward>(
-      f, *this, std::move(save_mean), std::move(save_std),
```

Comment:
Right.  This makes me realize there are probably two issues here:
1) The double backwards can probably be made more efficient by passing in the saved/running means/vars instead of recomputing
2) We don't actually test BatchNorm single or double backwards with training=False.